docs/HOL_Deploy_Private_Docker_Registry/README.md

```diff
@@ -0,0 +1,447 @@
+```
+$ sudo apt-get -y install apache2-utils
+```
+**Step 4.** [Docker Compose](https://docs.docker.com/compose/) allows you to write one YAML file to set up multiple containers and manage communication between them. Create a docker-compose.yml file using your favorite text editor:
+```
+$ mkdir ~/docker-registry && cd $_
+$ nano docker-compose.yml
+```
+
+**Step 5.** Set up a Docker registry container. Add and save the following contents to docker-compose.yml file, and replace **storage-account** and **storage-key** with your storage account name and access key:
+```
+registry:
+  image: registry:2
+  ports:
+    - 127.0.0.1:5000:5000
+  environment:
+    - REGISTRY_STORAGE=azure
+    - REGISTRY_STORAGE_AZURE_ACCOUNTNAME="<storage-account>"
+    - REGISTRY_STORAGE_AZURE_ACCOUNTKEY="<storage-key>"
+    - REGISTRY_STORAGE_AZURE_CONTAINER="registry"
+  volumes:
+    - ./data:/data
+```
+
+By hosting your Docker Registry instance on Azure VM and storing your images on Azure Blob Storage, you can have several benefits, such as security, performance and reliability.   
+
+**Step 6.** Set up a Docker Nginx container, and link it up to Docker registry container. Nginx is used to handle security and communication in this lab. Create a directory to store Nginx data:
+```
+$ mkdir ~/docker-registry/nginx
+```
+
+Re-open the docker-compose.yml file in the ~/docker-registry directory:
+```
+$ nano docker-compose.yml
+```
+
+Create a new Docker Nginx container based on the official Nginx image. Paste the following contents at the beginning of docker-compose.yml file and save it:
+```
+nginx:
+  image: "nginx:1.9"
+  ports:
+    - 443:443
+  links:
+    - registry:registry
+  volumes:
+    - ./nginx/:/etc/nginx/conf.d:ro
+```
+The interesting bit here is the **links** section. It automatically set up a "link" to the registry container. The **volumes** section provides a way to map files in Nginx container to the host machine.
+
+The final version of docker-compose.yml file should look similar to the following:
+
+```
+nginx:
+  image: "nginx:1.9"
+  ports:
+    - 443:443
+  links:
+    - registry:registry
+  volumes:
+    - ./nginx/:/etc/nginx/conf.d:ro
+registry:
+  image: registry:2
+  ports:
+    - 127.0.0.1:5000:5000
+  environment:
+    - REGISTRY_STORAGE=azure
+    - REGISTRY_STORAGE_AZURE_ACCOUNTNAME="mydockerstorageaccountname"
+    - REGISTRY_STORAGE_AZURE_ACCOUNTKEY="3c2BbH31iBjliyVh7CSjq9T/luFfwSt1zRZKXEWEu6HcImwXO9ccL124/MY4lemGGm8n7To2IVNr8EmRA=="
+    - REGISTRY_STORAGE_AZURE_CONTAINER="registry"
+  volumes:
+    - ./data:/data
+```
+
+**Step 7.** Create a registry.conf file:
+```
+$ nano ~/docker-registry/nginx/registry.conf
+```
+
+Copy and save the following contents into registry.conf file, and change the value of **server_name** to the DNS name or public IP of your Ubuntu VM (e.g., myregistrydomain.com):
```

Comment:
Yes, Julien. **myregistrydomain.com** is a bad example.  **<MyVM>.<MyRegion>.cloudapp.azure.com** is what I mean and used in the lab.